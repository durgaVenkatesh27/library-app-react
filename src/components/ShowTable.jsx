import React from 'react';
import * as _ from 'lodash';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { MaintainLibraryLocale } from '../locale';
import Button from '@material-ui/core/Button';

const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: "cornflowerblue",
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.action.hover,
    },
  },
}))(TableRow);

const ShowTable = props => {
  const { tableColumn, tableData, section, functionCall } = props;

  return (
    <div>
      <TableContainer component={Paper} >
        <Table aria-label="customized table" className={section}>
          <TableHead>
            <TableRow>
              {
                tableColumn.map((column, index) => (
                  <StyledTableCell
                    key={index}
                    align={"center"}
                    id={`${column}ColumnHeader`}
                  > {column}</StyledTableCell>
                ))
              }
            </TableRow>
          </TableHead>
          <TableBody>
            {_.isEmpty(tableData) ? " No data " : tableData.map((row, index) => (

              section === 'user' ? (
                <StyledTableRow key={row.id} id="tableBody">
                  <StyledTableCell align="center">{row.id}</StyledTableCell>
                  <StyledTableCell align="center">{row.name}</StyledTableCell>
                  <StyledTableCell align="center">{row.bookList}</StyledTableCell>
                </StyledTableRow>)
                : (section === 'book' ?
                  (<StyledTableRow key={row.id} id="tableBody">
                    <StyledTableCell align="center">{row.id}</StyledTableCell>
                    <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                  </StyledTableRow>)
                  : (section === 'borrow' ? (<StyledTableRow key={row.id} id="tableBody">
                    <StyledTableCell align="center">{row.id}</StyledTableCell>
                    <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                    <StyledTableCell align="center">
                      <Button
                        variant="contained"
                        size="small"
                        color="primary"
                        style={{ 'text-transform': 'none' }}
                        type="button"
                        className={`borrow${index} primary`}
                        onClick={(e) => functionCall(e, row.id)}
                        id={`borrow${index}`}
                      >
                        {_.get(MaintainLibraryLocale, 'borrowButtonTxt', '')}
                      </Button>
                    </StyledTableCell>
                  </StyledTableRow>)
                    : (<StyledTableRow key={row.id} id="tableBody">
                      <StyledTableCell align="center">{row.id}</StyledTableCell>
                      <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                      <StyledTableCell align="center">
                        <Button
                          variant="contained"
                          size="small"
                          color="secondary"
                          style={{ 'text-transform': 'none' }}
                          type="button"
                          onClick={(e) => functionCall(e, row.id)}
                          id={`borrowList${index}`}
                        >
                          {_.get(MaintainLibraryLocale, 'returnButtonTxt', '')}
                        </Button>
                      </StyledTableCell>
                    </StyledTableRow>)

                  ))
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}


export default ShowTable;