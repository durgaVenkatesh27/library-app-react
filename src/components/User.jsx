import React, { useState } from 'react';
import * as _ from 'lodash';
import { UserInfoLocale } from '../locale/index';
import ShowTable from './ShowTable';
import '../index.css';
import * as api from '../commons/Api';
import ErrorAlert from '../commons/ErrorAlert';


const User = props => {
  const { userInfo, updateUserData } = props;
  const [userName, setUserName] = useState('');
  const [alert, setAlert] = useState({ showMsg: false, message: '', severity: 'error' })

  const handleChange = (e) => {
    const { target: { value } } = e;
    alert.showMsg = false;
    setAlert(alert);
    setUserName(value);
  }

  const addUserInfo = (e) => {
    e.preventDefault();
    api.addUser(
      "http://localhost:9090/user/addUser",
      setUserName,
      { "name": userName },
      UserInfoLocale, setAlert);
    updateUserData();
  }

  return (
    <div>
      {
        _.get(alert, 'showMsg', false)
        && (<ErrorAlert errorMessage={_.get(alert, 'message', '')} errorSeverity={_.get(alert, 'severity', '')} />)
      }
      <form className="maintain-info">


        <label htmlFor="userName">
          {_.get(UserInfoLocale, 'userName', '')}
        </label>
        <input
          className="custom-input"
          type="text"
          name="userName"
          id="userName"
          value={userName}
          autoComplete='off'
          onChange={(e) => handleChange(e)}
        />
        <button
          className="custom-button"
          id="addUser"
          type="submit"
          disabled={userName.length === 0}
          onClick={(e) => addUserInfo(e)}
        >
          {_.get(UserInfoLocale, 'addButtonTxt', '')}
        </button>
      </form>
      <div>
        <ShowTable
          tableColumn={_.get(UserInfoLocale, 'userTableColumns', '')}
          tableData={userInfo}
          section='user'
        />
      </div>
    </div>
  );
}

export default User;