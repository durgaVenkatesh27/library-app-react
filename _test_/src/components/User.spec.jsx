import React from 'react';
import { mount, shallow } from 'enzyme';
import User from '../../../src/components/User';
import ShowTable from '../../../src/components/ShowTable';


describe('Home page Tests', () => {
    let props;
    global.fetch = jest.fn(() =>
        Promise.resolve({
            json: () => Promise.resolve({ rates: { CAD: 1.42 } }),
        }));
    beforeEach(() => {
        props = {
            userInfo: [{ id: 1, name: "sample", books: "", bookList: [] }],
            updateUserInfo: jest.fn()
        }
    })
    it('should render User Component', () => {
        const renderedModule = shallow(<User {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
    it('should render ShowTable Component', () => {
        const renderedModule = mount(<User {...props} />);
        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find(ShowTable).length).toBe(1);
    });
    it('should add user', () => {
        let user = [];
        const updateUserData = jest.fn(() => {
            user = [{ id: 1, name: 'Sample', books: "1", bookList: [1] },
            { id: 2, name: 'Sample1', books: "12", bookList: [12] }];
            return true;
        });
        const renderedModule = mount(<User {...props} updateUserData={updateUserData} />);

        renderedModule.find("#userName").simulate("change", {
            target: { value: "sample2" }
        });
        renderedModule.find("#addUser").simulate("click");

        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find('#userName').text()).toBe('');
    });
});