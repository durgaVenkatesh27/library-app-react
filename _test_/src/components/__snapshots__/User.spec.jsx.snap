// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Tests should add user 1`] = `ReactWrapper {}`;

exports[`Home page Tests should render ShowTable Component 1`] = `ReactWrapper {}`;

exports[`Home page Tests should render User Component 1`] = `ShallowWrapper {}`;
